File: EY.UbbstractThinkers.ProjectManagementPortal.Server/Models/Project.cs
using System;
using System.Collections.Generic;

namespace EY.UbbstractThinkers.ProjectManagementPortal.Server.Models
{
    public class Project
    {
        public Guid Uid { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public DateOnly StartDate { get; set; }
        public DateOnly? EndDate { get; set; }

        public List<ProjectStakeholder> Stakeholders { get; set; } = [];
        public List<ProjectResources> Resources { get; set; } = [];
        public List<ApprovalRequest> Approvals { get; set; } = [];
        public string OwnerId { get; set; }

        public Guid TemplateUid { get; set; }
        public Template Template { get; set; }

        public Guid CurrentStageUid { get; set; }
        public Stage CurrentStage { get; set; }
    }
}

File: ey.ubbstractthinkers.projectmanagementportal.client/src/models/Project.ts
import type { Dayjs } from "dayjs";
import dayjs from "dayjs";

export interface Project {
  uid?: string;
  name: string;
  description: string;
  startDate: Dayjs | string;
  endDate: Dayjs | string | null;
  ownerId: string | null;
  stakeholderIds: string[];
  resourceIds: string[];
  templateUid: string;
  currentStageUid?: string;
}

export const DefaultProject: Project = {
  name: "",
  description: "",
  startDate: dayjs(),
  endDate: null,
  ownerId: null,
  stakeholderIds: [],
  resourceIds: [],
  templateUid: "",
};
